crates/turborepo/tests/common/mod.rs

```diff
@@ -0,0 +1,56 @@
+use std::{path::Path, process::Command};
+
+use camino::Utf8Path;
+
+pub fn setup_fixture(
+    fixture: &str,
+    package_manager: Option<&str>,
+    test_dir: &Path,
+) -> Result<(), anyhow::Error> {
+    let script_path = Utf8Path::new(env!("CARGO_MANIFEST_DIR"))
+        .join("../../turborepo-tests/helpers/setup_integration_test.sh");
+
+    Command::new("bash")
+        .arg("-c")
+        .arg(format!(
+            "{} {} {}",
+            script_path,
+            fixture,
+            package_manager.unwrap_or("npm@10.5.0")
+        ))
+        .current_dir(test_dir)
+        .spawn()?
+        .wait()?;
+
+    Ok(())
+}
+
+/// Executes a command with different arguments in a specific fixture and
+/// package manager. Creates a snapshot file for each set of arguments
+#[macro_export]
+macro_rules! check {
+    ($fixture:expr, $package_manager:expr, $command:expr, $($name:expr => $query:expr,)*) => {
+        {
+            let tempdir = tempfile::tempdir()?;
+            crate::common::setup_fixture($fixture, None, tempdir.path())?;
+            $(
+                let output = assert_cmd::Command::cargo_bin("turbo")?
+                .arg($command)
+                    .arg($query)
+                    .current_dir(tempdir.path())
+                    .output()?;
+
+                let stdout = String::from_utf8(output.stdout)?;
+                let query_output: serde_json::Value = serde_json::from_str(&stdout)?;
+                let test_name = format!(
+                    "{}_{}_{}_({})",
+                    module_path!(),
+                    $fixture,
+                    $name.replace(' ', "_"),
+                    $package_manager
+                );
+                insta::assert_json_snapshot!(test_name, query_output);
```

Comment:
Can we put each of these asserts in it's own test? Currently if the first test case fails, none of the other ones will run. Prysk will continue running if one of the commands doesn't produce matching output.

Reply:
Is there a way to do this while also using the same fixture? I don't want to spin up a fixture for each test. Insta does also let you do the prysk behavior with `cargo insta test --review` or `INSTA_FORCE_PASS=1 cargo test --no-fail-fast`

Reply:
I suppose we'd have to store the tempfile in something like a static ref and have special cleanup logic. I think eventually we'll want `beforeAll`/`afterAll` behavior to make debugging failures easier. ITG

Reply:
> Insta does also let you do the prysk behavior with cargo insta test --review or INSTA_FORCE_PASS=1 cargo test --no-fail-fast

I don't think these do what we want, `INSTA_FORCE_PASS=1` seems to just forcibly pass all snapshot assertions without printing any diffs. `cargo insta test --review` is just equivalent to `--interactive` in prysk.